internal/service/alarms/internal/middleware.go

```diff
@@ -0,0 +1,54 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/getkin/kin-openapi/openapi3filter"
+	oapimiddleware "github.com/oapi-codegen/nethttp-middleware"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/api/generated"
+)
+
+type Middleware = func(http.Handler) http.Handler
+
+// LogDuration log time taken to complete a request.
+// TODO: This is just get started with middleware but should be replaced with something that's more suitable for production i.e OpenTelemetry
+// https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/examples/prometheus/main.go
+func LogDuration() Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			startTime := time.Now()
+			next.ServeHTTP(w, r)
+			fmt.Printf("%q took %s\n", r.RequestURI, time.Since(startTime))
```

Comment:
These errors should go through a logging framework.

Reply:
logging instead of printing everywhere now.